packages/package-json/src/index.js

```diff
@@ -1,358 +1,252 @@
+	 *
+	 * @returns {object}
+	 */
+	function getDocument() {
+		return deepCloneObject(workingContents);
+	}
+
+	/**
+	 * Check if there are file changes to write.
+	 *
+	 * @returns boolean
+	 */
+	function hasChangesToWrite() {
+		const formattedPreviousContents = formatObjectAsJson(previousContents);
+		const formattedWorkingContents = formatObjectAsJson(workingContents);
+
+		return formattedPreviousContents !== formattedWorkingContents;
+	}
+
+	/**
+	 * Write to the `package.json` file.
+	 *
+	 * @returns boolean
+	 */
+	function writeChanges() {
+		try {
+			fs.writeFileSync(
+				options.filepath,
+				formatObjectAsJson(workingContents) + "\n"
+			);
+		} catch (err) {
+			throw new Error(
+				`package-json#writeChanges: Error writing changes to file '${
+					options.filepath
+				}'\n\n${err.message}`
+			);
+		}
+
+		previousContents = deepCloneObject(workingContents);
+
+		return true;
+	}
+
+	/**
+	 * Get a specific field from the `package.json` object.
+	 *
+	 * @param {string} field
+	 * @returns {*}
+	 */
+	function getField(field) {
+		return workingContents[field];
+	}
+
+	/**
+	 * Set the value for a specific field in the `package.json` object.
+	 *
+	 * @param {string} field
+	 * @param {*} value
+	 *
+	 * @returns {object} - changelog entry
+	 */
+	function setField(field, value) {
+		const changes = { event: "setField", field };
+
+		const fieldAlreadyExists = typeof workingContents[field] !== "undefined";
+
+		if (fieldAlreadyExists) {
+			changes.previousValue = workingContents[field];
+		}
+
+		workingContents[field] = value;
+
+		return changelog.createEntry(changes);
+	}
+
+	/**
+	 * Require a package to exist as a dependency in `package.json`.
+	 *
+	 * @param {object} options
+	 * @param {string} options.pkg - Package name
+	 * @param {string} options.version
+	 * @param {string} options.field
+	 *
+	 * @returns {object} - changelog entry
+	 */
+	function requireDependency({ pkg, version, field }) {
+		if (!dependencyFields.includes(field)) {
+			throw new Error(
+				`package-json#requireDependency: Invalid field specified '${field}'. Valid fields: ${dependencyFields.join(
+					", "
+				)}`
+			);
+		}
+
+		if (!workingContents[field]) {
+			workingContents[field] = {};
+		}
+
+		const dependencies = workingContents[field];
+
+		const changes = { event: "requireDependency", field, pkg, version };
+
+		const dependencyAlreadyExists = typeof dependencies[pkg] !== "undefined";
+		changes.alreadyExisted = dependencyAlreadyExists;
+		if (dependencyAlreadyExists) {
+			changes.previousValue = dependencies[pkg];
+		}
+
+		dependencies[pkg] = version;
+
+		return changelog.createEntry(changes);
+	}
+
+	/**
+	 * Remove a package as a dependency from `package.json`.
+	 *
+	 * @param {object} options
+	 * @param {string} options.pkg
+	 * @param {string} options.field
+	 *
+	 * @returns {object|boolean} - changelog entry or `false` if dependency doesn't exist
+	 */
+	function removeDependency({ pkg, field }) {
```

Comment:
```suggestion
	function removeDependency({ package, field }) {
```

This seems like an unnecessary abbreviation?

Reply:
Annoyingly, `package` is a reserved keyword in the ECMAScript specification: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Future_reserved_keywords

If you name something `package`, ESLint (rightly) raises an error.

Reply:
I'm adding comments in the module to explain this.